src/Launcher/launcher_1l.jl

```diff
@@ -128,16 +160,21 @@ function launch_1l!(
         else
             # update step size
             dΛ = max(bmin, min(bmax * Λ, 0.85 * (1.0 / err)^(1.0 / 3.0) * dΛ))
-            dΛ = min(dΛ, Λ - Λf)
+
+            # check if we pass by required checkpoint and adjust step size accordingly
+            cps_lower = filter(x -> x < 0.0, cps .- Λ)
+            Λ_cp      = 0.0
+
+            if length(cps_lower) >= 1
+                Λ_cp = Λ + first(cps_lower)
+            end
+
+            dΛ = min(dΛ, Λ - Λ_cp)
```

Comment:
I think here we do not need to check if we pass a checkpoint, since at the start and after each successful step we already ensure that we do not pass a checkpoint and the correction here will only make the step smaller.

Reply:
In fact, since the check is only for the next ODE step and only the current one, this needs to stay in order to really pass by the cps which are required.

Reply:
I agree with @lgresista: dLambda at that point was already chosen, such that we do not step over the next cp. the new dLambda will by construction always be small than the rejected one, i.e. we for sure will not step over a cp